src/crypto/tests/TestPersistentStorageOpKeyStore.cpp

```diff
@@ -33,6 +33,99 @@ using namespace chip::Crypto;
 
 namespace {
 
+/**
+ * Implementation of OperationalKeystore for testing purposes.
+ *
+ * Not all methods of OperationalKeystore are implemented, only the currently needed.
+ * Currently this implementation supports only one fabric and one operational key.
+ *
+ * The Validate method can be used to validate the provided P256SerializedKeypair with the
+ * stored kP256SerializedKeypairRaw data.
+ */
+class TestOperationalKeystore final : public Crypto::OperationalKeystore
+{
+public:
+    constexpr static uint8_t kP256SerializedKeypairRaw[] = {
+        0x4,  0xd0, 0x99, 0xde, 0xd1, 0x15, 0xea, 0xcf, 0x8f, 0x13, 0xde, 0xaf, 0x74, 0x65, 0xf3, 0x10, 0x3a, 0x75, 0x94, 0x51,
+        0x37, 0x3c, 0xc,  0x9a, 0x25, 0xc7, 0xad, 0xb4, 0x31, 0x39, 0x62, 0xec, 0x12, 0xa3, 0xdf, 0x28, 0x5f, 0x2c, 0x86, 0x47,
+        0x2d, 0x1f, 0x5d, 0x45, 0x1d, 0x9f, 0xbc, 0xe8, 0x47, 0xf2, 0x1f, 0x40, 0x17, 0x61, 0x2b, 0x9a, 0x4e, 0x68, 0x9c, 0xe9,
+        0x9e, 0xb7, 0x45, 0xdc, 0xcd, 0xb,  0x90, 0xd0, 0x24, 0xa5, 0x6d, 0x64, 0x97, 0x62, 0x75, 0x42, 0x91, 0x74, 0xfc, 0xfe,
+        0xcb, 0x1,  0x6c, 0xc,  0x74, 0x6f, 0x39, 0x9f, 0x5,  0x96, 0x1b, 0xe6, 0x4a, 0x97, 0xe5, 0x84, 0x72
+    };
+
+    bool HasOpKeypairForFabric(FabricIndex fabricIndex) const override { return fabricIndex != mUsedFabricIndex; };
+
+    CHIP_ERROR NewOpKeypairForFabric(FabricIndex fabricIndex, MutableByteSpan & outCertificateSigningRequest) override
+    {
+        // Only one Fabric is supported
+        if (mUsedFabricIndex != 0)
+        {
+            ChipLogError(Test,
+                         "The TestOperationalKeystore has been initialized already, please use RemoveOpKeypairForFabric or remove "
+                         "the object to store a new fabric");
+            return CHIP_ERROR_INVALID_FABRIC_INDEX;
+        }
+        mUsedFabricIndex = fabricIndex;
+        (void) outCertificateSigningRequest;
+
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR ExportOpKeypairForFabric(FabricIndex fabricIndex, Crypto::P256SerializedKeypair & outKeypair) override
+    {
+        // Simulate not existing value
+        if (fabricIndex != mUsedFabricIndex)
+        {
+            return CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
```

Comment:
According to your doc in `OperationalKeystore.h`, this should return CHIP_ERROR_INVALID_FABRIC_INDEX.